tests/kernel/gen_isr_table/src/main.c

```diff
@@ -136,8 +136,15 @@ void isr6(void *param)
  * doesn't support this; we need to tell the compiler not to reorder memory
  * accesses to trigger_check around calls to trigger_irq.
  */
+#if defined(__CCAC__)
+/* MWDT does not support _attribute__((optimize("-O0"))), buf the following
+ * it equals -Os
```

Comment:
This function is a confusing mess, and IMHO this makes it worse.  Really someone should come in there and implement the ordering constraints described in a portable way (e.g. call stuff through a C function pointer and not compiler voodoo, whatever).

But barring that, please make this a NO_OPTIMIZE macro or something and declare it at the toolchain level.